main.tf

```diff
@@ -27,18 +27,24 @@ resource "random_string" "ipsec_psk" {
 resource "metal_device" "router" {
   hostname         = var.hostname
   plan             = var.plan
-  facilities       = [var.facility]
+  metro            = var.metro
   operating_system = var.operating_system
   billing_cycle    = var.billing_cycle
   project_id       = var.project_id
   ipxe_script_url  = var.ipxe_script_url
   always_pxe       = var.always_pxe
 }
 
-resource "metal_port_vlan_attachment" "router_vlan_attach" {
+resource "metal_device_network_type" "convert-networking" {
```

Comment:
thoughts on converting this to metal_port?

At a glance, I think it would be:
```hcl
resource "metal_port" "bond0" {
  port_id = [for p in metal_device. router.ports: p.id if p.name == "bond0"][0]
  layer2 = false
  bonded = false
  vxlan_ids = [metal_vlan.private_vlan.vxlan]
}
```

Reply:
I personally find the network type and vlan attachment method simpler and more aligned with a user's experience in the portal. If this is meant to be deprecated then I can switch over to the metal_port method.

Reply:
I see the metal_port resource as a more advanced option that behaves similar to manual API usage.